tests/test_vera.py

```diff
@@ -285,7 +286,11 @@ def test_vera_different_shapes(self, mlp):
     @pytest.mark.parametrize("dtype", [torch.float32, torch.float16, torch.bfloat16])
     def test_vera_dtypes(self, dtype):
         # 1872
-        if (dtype == torch.bfloat16) and not (torch.cuda.is_available() and torch.cuda.is_bf16_supported()):
+        if (
+            (dtype == torch.bfloat16)
+            and not (torch.cuda.is_available() and torch.cuda.is_bf16_supported())
+            and infer_device() != "xpu"
```

Comment:
The condition is getting hard to read, how about breaking it down like this?

```python
        if dtype == torch.bfloat16:
            # skip if bf16 is not supported on hardware, see #1872
            is_xpu = infer_device == "xpu"
            is_cuda_bf16 = torch.cuda.is_available() and torch.cuda.is_bf16_supported()
            if not (is_xpu or is_cuda_bf16):
                pytest.skip("bfloat16 not supported on this system, skipping the test")

```

Reply:
Okay, do you want to update the PR accordingly?

Reply:
yes, just a second!